From 5ef9c848019ed840c8bc8bf10161f0464c50bcab Mon Sep 17 00:00:00 2001
From: Pengyu Luo <mitltlatltl@gmail.com>
Date: Sun, 26 Jul 2026 15:28:13 +0800
Subject: [PATCH 1/2] caihong: overclock the mdp clock

This only makes clock rate 515935728 legal and increase little power for the
stability when clk hits the rate , otherwise this mode will be rejected due to
MODE_CLOCK_HIGH. If 4 LMs / quad-pipeline is supported, then we can drop this. 

515935728 = (1500 + 118 + 20 + 100) * 2 * (2120 + 26 + 2 + 208) * 120 * 1.05 / 2
---
 arch/arm64/boot/dts/qcom/sm8650.dtsi | 5 +++++
 drivers/clk/qcom/dispcc-sm8550.c     | 1 +
 2 files changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sm8650.dtsi b/arch/arm64/boot/dts/qcom/sm8650.dtsi
index 9ecbc0010694..63a4728f8c41 100644
--- a/arch/arm64/boot/dts/qcom/sm8650.dtsi
+++ b/arch/arm64/boot/dts/qcom/sm8650.dtsi
@@ -5751,6 +5751,11 @@ opp-514000000 {
 						opp-hz = /bits/ 64 <514000000>;
 						required-opps = <&rpmhpd_opp_nom>;
 					};
+
+					opp-600000000 {
+						opp-hz = /bits/ 64 <600000000>;
+						required-opps = <&rpmhpd_opp_nom_l1>;
+					};
 				};
 			};
 
diff --git a/drivers/clk/qcom/dispcc-sm8550.c b/drivers/clk/qcom/dispcc-sm8550.c
index 418dcea20f00..b98cafd1978b 100644
--- a/drivers/clk/qcom/dispcc-sm8550.c
+++ b/drivers/clk/qcom/dispcc-sm8550.c
@@ -608,6 +608,7 @@ static const struct freq_tbl ftbl_disp_cc_mdss_mdp_clk_src_sm8650[] = {
 	F(325000000, P_DISP_CC_PLL0_OUT_MAIN, 3, 0, 0),
 	F(402000000, P_DISP_CC_PLL0_OUT_MAIN, 3, 0, 0),
 	F(514000000, P_DISP_CC_PLL0_OUT_MAIN, 3, 0, 0),
+	F(600000000, P_DISP_CC_PLL0_OUT_MAIN, 3, 0, 0),
 	{ }
 };
 
-- 
2.54.0

